components/header.tsx

```diff
@@ -56,29 +55,29 @@ export default function Header() {
         }}
       >
         <Link href="/" passHref>
-          <NavLink sx={mobileLinksStyle}>Home</NavLink>
+          <NavLink onClick={toggleMenu} sx={mobileLinksStyle}>Home</NavLink>
```

Comment:
Ok so replicating `useEffects` solution also inside _header_ seems to be the best solution.

```  useEffect(() => {
    router.events.on("routeChangeComplete", toggleMenu);
    return () => {
      router.events.off("routeChangeComplete", toggleMenu);
    };
  }, [router.events]);
```

By doing so, if we click the same link of the current route nothing happens.
The menu also remain opened.

Reply:
but I want to close the menu, so to be honest I'm not sure but what should the user expect. If a user press a button I think it's better to always hide the menu because the user I think expect to see the page after an action, if this doesn't happen I mean maybe the user realize the mistake and press the menu button to close but still we are forcing the user to do something while if we handle this here it will just work fine, what do you think?